src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePlugin.java

```diff
@@ -212,23 +224,57 @@ public void deleteInstance(ComputeOrder order, CloudStackUser cloudUser) throws
         return null;
     }
 
-    private GetAllServiceOfferingsResponse getServiceOfferings(CloudStackUser cloudUser) throws FogbowException {
+    @VisibleForTesting
+    List<GetAllServiceOfferingsResponse.ServiceOffering> filterServicesOfferingByRequirements(
+            final List<GetAllServiceOfferingsResponse.ServiceOffering> serviceOfferings,
+            final ComputeOrder computeOrder) {
+
+        List<GetAllServiceOfferingsResponse.ServiceOffering> serviceOfferingsFilted = serviceOfferings;
+        Map<String, String> requirements = computeOrder.getRequirements();
+        if (requirements == null && requirements.size() == 0) {
+            return serviceOfferings;
+        }
+
+        for (Map.Entry<String, String> tag : requirements.entrySet()) {
+            String tagFromRequirements = tag.getKey() + FOGBOW_TAG_SEPARATOR + tag.getValue();
+            serviceOfferingsFilted = serviceOfferingsFilted.stream().filter(serviceOffering -> {
+                String tagsServiceOffering = serviceOffering.getTags();
+                boolean isMatchingWithRequirements = tagsServiceOffering != null &&
+                        !tagsServiceOffering.isEmpty() &&
+                        tagsServiceOffering.contains(tagFromRequirements);
+                return isMatchingWithRequirements;
+            }).collect(Collectors.toList());
+        }
+
+        return serviceOfferingsFilted;
+    }
+
+    @VisibleForTesting
+    GetAllServiceOfferingsResponse getServiceOfferings(final CloudStackUser cloudUser) throws FogbowException {
+        if (cloudUser == null) {
+            throw new FogbowException(CLOUDUSER_NULL_EXCEPTION_MSG);
+        }
+
         GetAllServiceOfferingsRequest request = new GetAllServiceOfferingsRequest.Builder().build(this.cloudStackUrl);
         CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
 
         String jsonResponse = null;
+        GetAllServiceOfferingsResponse getAllServiceOfferingsResponse = null;
         try {
-            jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
+            jsonResponse = doGet(request.getUriBuilder().toString(), cloudUser);
+            getAllServiceOfferingsResponse = GetAllServiceOfferingsResponse.fromJson(jsonResponse);
         } catch (HttpResponseException e) {
             CloudStackHttpToFogbowExceptionMapper.map(e);
         }
 
-        GetAllServiceOfferingsResponse serviceOfferingsResponse = GetAllServiceOfferingsResponse.fromJson(jsonResponse);
-
-        return serviceOfferingsResponse;
+        return getAllServiceOfferingsResponse;
     }
 
-    private GetAllDiskOfferingsResponse.DiskOffering getDiskOffering(int diskSize, CloudStackUser cloudUser) throws FogbowException {
+    @Nullable
+    @VisibleForTesting
+    GetAllDiskOfferingsResponse.DiskOffering getDiskOffering(int diskSize, CloudStackUser cloudUser)
```

Comment:
In coverage verification this method does not appear to be covered by testing, make sure that it does.

Reply:
I've tested this method, but I will verify again.

Reply:
Done